### ISSUE TITLE
CompleteRequestProto Not Set FileId Cause FileWriter.Close Error

### ISSUE THREAD

**Issue by 982945902** [colinmarc/hdfs#298]
that is failed stack:
![截屏2022-08-09 下午8 02 56](https://user-images.githubusercontent.com/49055103/183642734-f26bacf0-5e1c-4d62-b9e0-ce74d444182d.png)

under the circumstances，use "hdfs dfs -ls" show file size is zero，but use "hdfs dfs -head" can normally display file content, and use "hdfs dfs copyToLoacl" ,the downloaded file is  normally

i refer to ‘hdfs dfs’ call sequence , before call 'complete', i call 'getFileInfo' to get 'FileId', and set to CompleteRequestProto, like this:
```
func (f *FileWriter) Close() error {
	var lastBlock *hdfs.ExtendedBlockProto
	if f.blockWriter != nil {
		lastBlock = f.blockWriter.Block.GetB()

		// Close the blockWriter, flushing any buffered packets.
		err := f.finalizeBlock()
		if err != nil {
			return err
		}
	}

	var err error
	getFileInfoReq := &hdfs.GetFileInfoRequestProto{Src: proto.String(f.name)}
	getFileInfoResp := &hdfs.GetFileInfoResponseProto{}

	err = f.client.namenode.Execute("getFileInfo", getFileInfoReq, getFileInfoResp)
	if err != nil {
		return &os.PathError{"getFileInfo", f.name, err}
	}

	completeReq := &hdfs.CompleteRequestProto{
		Src:        proto.String(f.name),
		ClientName: proto.String(f.client.namenode.ClientName),
		Last:       lastBlock,
		FileId:     getFileInfoResp.Fs.FileId,
	}
	completeResp := &hdfs.CompleteResponseProto{}

	err = f.client.namenode.Execute("complete", completeReq, completeResp)
	if err != nil {
		return &os.PathError{"create", f.name, err}
	} else if completeResp.GetResult() == false {
		return &os.PathError{"create", f.name, ErrReplicating}
	}

	return nil
}
```
after changed code, i try again, successfully fix 'FileWriter.Close Error'

because of i do not known much about hdfs,  please check the cause of the problem.

**Comment by 982945902** [comment]
```
under the circumstances，use "hdfs dfs -ls" show file size is zero，but use "hdfs dfs -head" can normally display file content, and use "hdfs dfs copyToLoacl" ,the downloaded file is normally
```
after several hours, try 'hdfs dfs -ls' again, this file size is normally,  so i guess if ‘FileId’ effect hdfs sync

**Comment by colinmarc** [comment]
I think this was fixed by #301, but please re-open if that's not the case.
